qa/tasks/cephfs/caps_helper.py

```diff
@@ -43,6 +43,15 @@ def write_test_files(self, mounts, testpath=''):
             dirpath = os_path_join(mount_x.hostfs_mntpt, testpath, dirname)
             mount_x.run_shell(f'mkdir {dirpath}')
             filepath = os_path_join(dirpath, filename)
+            # XXX: the reason behind adding filepathm, cephfs_name and both
```

Comment:
I'm not 100% clean on how this catches bugs -- is there a verification of any sort somewhere?

Reply:
> I'm not 100% clean on how this catches bugs -- is there a verification of any sort somewhere?

Yes. There is a verification.

The problem is that if file data is also identical for all files created for testing across all CephFSs (say `cephfs1` and `cephfs2` in this case) it's possible for a test case meant for `cephfs1` to pass even when it's looking at a file from `cephfs2` (instead of `cephfs1`).

The verification is that code meant for testing access levels (`r` and `rw`) look at the file data just obtained from read_file() and at the file data saved in `filedata` to make sure that it is looking at right file. Copying from `conduct_pos_test_for_read_caps()` and `conduct_pos_test_for_write_caps()` -

```
self.assertEqual(data, contents) # data is from filedata and contents is obtained from read_file()
```

Reply:
OK - makes sense.